pkg/lib/interaction/nodes/authn_oob.go

```diff
@@ -36,7 +40,12 @@ func (e *EdgeAuthenticationOOB) Instantiate(ctx *interaction.Context, graph *int
 		},
 	})
 	if err != nil {
-		info = nil
```

Comment:
What is the reason we should handle the error in `NodeAuthenticationEnd`?

Reply:
Per offline discussion:
1. We should return error directly instead of handling the error in `NodeAuthenticationEnd`.
1. We should dispatch AuthenticationFailedEvent in `Authenticators.VerifyWithSpec()` which is always called when authentication.

Reply:
@louischan-oursky I found that there will be some problem if we dispatch the event in `VerifyWithSpec`, because for some authenticators, such as passkey and totp, the user can have multiple authenticators. In current implementation, we will test all authenticators with `VerifyWithSpec` and therefore causing the event dispatched multiple times. (Yes, lockout and ratelimit should have the same problem, therefore I think I will create another issue to fix that).
In this pr, maybe I simply delegate the error to NodeAuthenticationEnd like before.